### PR TITLE
ServiceMonitor resource for Prometheus scraping

### PR DESCRIPTION
**Pull request by nlamirault**
# Pull Request

## Related issue

- [#3523](https://github.com/meilisearch/meilisearch/issues/3523)

## What does this PR do?
- Creates a ServiceMonitor Kubernetes resources.

Prometheus metrics must be enabled using the `--experimental-enable-metrics` CLI argument

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Comment by brunoocasali** [comment]
@alallema I think this PR will be necessary to the adoption of the new experimental feature https://github.com/meilisearch/meilisearch/issues/3523

So what I suggest you to do is:

- [ ] Fix parameters list like you suggested to the contributor
- [ ] Add comments and point to the engine issue regarding this experimental feature.
- [ ] Add support to Authorization token when a master key protects the instance.

**Comment by alallema** [comment]
Hi @nlamirault,
Thank you so much for continuing to update your code each time ❤️ . I didn't forget you. I will finish by merging your PR, I just need to check some stuff before and I still don't have time to do it, but I will.
Thank you so much for your patience 😊

**Comment by alallema** [comment]
Hi @nlamirault,
Sorry to come back to this PR so late. I finally checked it.
Rather than using the CLI command `--experimental-enable-metrics`, would it be possible to add the `MEILI_EXPERIMENTAL_ENABLE_METRICS: "false"` parameter to the environment variables and then if `{{- if.Values.serviceMonitor.enabled -}}` set it as `(eq.Values.environment.MEILI_EXPERIMENTAL_ENABLE_METRICS "true")`
I'm not an expert, so I'd like your opinion. I can implement it in a subsequent PR if it's annoying.
Thank you 😊

File: charts/meilisearch/templates/serviceMonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+spec:
+  jobLabel: {{ .Release.Name }}
+  namespaceSelector:
+    matchNames:
+      - {{ .Release.Namespace }}
+  selector:
+    matchLabels:
+      {{- include "meilisearch.selectorLabels" . | nindent 6 }}
+  endpoints:
+    - port: http
```
Comment (mike-stewart) [inline charts/meilisearch/templates/serviceMonitor.yaml:20]: Haven't tested this myself yet, but I was looking at adding a ServiceMonitor to my deployment, and I think you'd need something like this to support authentication, right?
```yaml
  endpoints:
    - bearerTokenSecret:
        name: meilisearch-master-key
        key: MEILI_MASTER_KEY
...
```

**Review by alallema — APPROVED** [review]

**Comment by alallema** [comment]
bors merge

**Pull request merged**